gengokit/generator/gen_test.go

```diff
@@ -436,46 +427,53 @@ func renderService(svc *svcdef.Service, prev string, te *gengokit.TemplateExecut
 		return "", err
 	}
 
-	nextBytes, err = testFormat(nextBytes)
+	nextCode, err := testFormat(string(nextBytes))
 	if err != nil {
 		return "", errors.Wrap(err, "cannot format")
 	}
 
-	nextCode := strings.TrimSpace(string(nextBytes))
+	nextCode = strings.TrimSpace(nextCode)
 
 	return nextCode, nil
 }
 
-func testGenerateResponseFile(templFP string, te *gengokit.TemplateExecutor, prevGenMap map[string]io.Reader) ([]byte, error) {
+func diff(a, b string) string {
+	return gentesthelper.DiffStrings(
+		a,
+		b,
+	)
+}
+
+func testGenerateResponseFile(executor *gengokit.Executor, prev io.Reader, templFP string) (string, error) {
```

Comment:
This is a bit of a nit, but as a reader it tripped me up ever so slightly. This is a little weird because it's a helper function but its return type is not the same as `generateResponseFile`, which returns a `NamedReader` or something like that. This sort of creates some dissonance between the implementation and the test, given the similarity of the names. Perhaps give the test function a distinct name. Of course, I wasn't crazy about `generateResponseFile` to begin with, but the point stands regardless of the name of the non-test function.

Further, this test function appears to be doing redundant work--it reads the generated code, then formats it. This happens within `generateResponseFile` though. If there's a reason for it being done again here, it's not obvious without really digging in.